api/src/license_serializer.rs

```diff
@@ -0,0 +1,46 @@
+#![allow(arithmetic_overflow)]
+
+use std::num::Wrapping;
+
+pub trait LicenseKeySerializer {
```

Comment:
You does not require that `&self` in your DefaultKeySerializer. So it can be converted to static functions.